Cython/Utility/ModuleSetupCode.c

```diff
@@ -381,6 +381,12 @@
   #endif
 #endif
 
+#if CYTHON_USE_MODULE_STATE && CYTHON_PEP489_MULTI_PHASE_INIT
+#error "Cannot combine CYTHON_USE_MODULE_STATE and CYTHON_PEP489_MULTI_PHASE_INIT"
+/* since PyState_FindModule requires that each module-def is linked to 1 (or 0) modules
+and multi-phase init allows the same module to be imported many times */
+#endif
+
```

Comment:
Should this really be a compile time error? The fact that is can be imported multiple times does not mean that it will be.

Reply:
[PyState_AddModule](https://docs.python.org/3/c-api/module.html#c.PyState_AddModule) is documented as not working on multi-phase modules (and does appear to check that at runtime) so I believe it genuinely won't work at all, and hence probably is worth a compile-time check.

Of course long-term we'd want an implementation of `CYTHON_USE_MODULE_STATE` that does work with multi-phase modules.

Reply:
With that said, this isn't necessary for the rest of this PR so I can happily get rid of it.

Reply:
Let's remove it for now. We can add a stricter (runtime?) check later.
```suggestion
```